doc/getstarted/build_and_install/docker_install_en.rst

```diff
@@ -8,199 +8,236 @@ Please be aware that you will need to change `Dockers settings
 <https://github.com/PaddlePaddle/Paddle/issues/627>`_ to make full use
 of your hardware resource on Mac OS X and Windows.
 
+Working With Docker
+-------------------
+
+Here we will describe the basic docker concepts that we will be using
+in this tutorial.
+
+- *container* is an environment for running applications
```

Comment:
It would be easier to describe *image* before *container*, as an image is a program and a container is an instance.   Also, it might be helpful to add a paragraph describing why we use Docker.  I propose the following:

-----

- *image*: A Docker image is a pack of software. It could contain one or more programs and all their dependencies.  For example, the PaddlePaddle's Docker image includes pre-built PaddlePaddle and Python and many Python packages.  We can run a Docker image directly, other than installing all these software.  We can type 

   ```bash
   docker images
   ```

   to list all images in the system.  We can also run 

   ```bash
   docker pull paddlepaddle/paddle
   ```

   to download a Docker image, `paddlepaddle/paddle` in this example, from Dockerhub.com.

- *container*: considering a Docker image a program, a container is a "process" that runs the image.  Indeed, a container is exactly an operating system process, but with a virtualized filesystem, network port space, and other virtualized environment.  We can type 

   ```bash
   docker run paddlepaddle/paddle
   ```

   to start a container to run a Docker image, `paddlepaddle/paddle` in this example.